### ISSUE TITLE
COPY errors result in hangs or delayed exceptions

### ISSUE THREAD

**Issue by adam-nielsen** [brianc/node-pg-copy-streams#125]
If there is an error in the `COPY`, the error handler does not properly cancel the query.  This has two effects - if a query timeout is specified, then there is an undefined function error upon the timer expiring, and if no query timeout is specified, then the code hangs forever after the error is thrown.

The problem seems to be because a timer is created in the query() function, but it is not cancelled by node-pg-copy-streams upon encountering an error.

Here is an example that demonstrates the problem.  When run with a timeout (as shown), it results in this error:

    TypeError: queryCallback is not a function
        at Timeout._onTimeout (node_modules/pg/lib/client.js:532:9)

The code is:

```
const pg = require('pg');
const pgs = require('pg-copy-streams');

async function run()
{
	const db = new pg.Client({
		// With this line, a JS error happens after two seconds.
		// If the line is commented out, the program will never exit after the error.
		query_timeout: 2000,
	});
	await db.connect();
	console.log('Connected');

	await new Promise((resolve, reject) => {
		const query = `CREATE TEMP TABLE "t" (n INTEGER) ON COMMIT DROP; COPY "t" FROM STDIN;`;
		const s = db.query(pgs.from(query));
		s.on('finish', () => {
			console.log('finish()');
			resolve();
		});
		s.on('error', e => {
			console.error('Received error:', e);
			reject(e);
		});
		s.write('A'); // cause an error
		s.end();
	});

	console.log('Done');
}

async function main()
{
	try {
		await run();
	} catch (e) {
		console.error('Caught', e.message);
	}

	console.log('Waiting for any timeouts to expire...');
	await new Promise((resolve, reject) => {
		setTimeout(resolve, 10000);
	});
	console.log('Exiting');
}

main();
```

**Comment by jeromew** [comment]
Thanks for your work on this and pointing to a solution.

I must admit that the out-of-band timeout mechanism provided by pg is currently not tested in the test suite (I never used it and it adds a level of complexity to the error case scenario)

I will take a look at it and try to understand the spirit of the implementation and of the discussion on https://github.com/brianc/node-postgres/pull/2614

**Comment by jeromew** [comment]
I just published version 6.0.2 which should fix the issue regarding the interaction with the timeout mechanism. It does not need the fix in `pg` regarding the empty callback. I close this issue for now.

**Comment by adam-nielsen** [comment]
Perfect!  This has fixed the problem, many thanks!
